x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/results/ErrorChunkedInferenceResults.java

```diff
@@ -94,4 +97,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
     public String getWriteableName() {
         return NAME;
     }
+
+    @Override
+    public Iterator<Chunk> chunksAsMatchedTextAndByteReference(XContent xcontent) {
+        return List.of(exception).stream().map(e -> new Chunk(e.getMessage(), (BytesReference) null)).iterator();
```

Comment:
it should be ok to return a null bytes reference, right?

Reply:
Yes - SemanticText is not using this implementation, just checking when it's receiving an instance of the class to mark the inference as failed.

We could as well throw an `UnsupportedOperationException` for this

Reply:
I changed it to use `BytesArray::EMPTY` as I'm not sure what the benefit would be of throwing an exception instead, but I'm open to that option.